linkerd/app/outbound/src/http/logical/policy/route.rs

```diff
@@ -149,15 +165,6 @@ impl<T: Clone, M, F, P> svc::Param<BackendDistribution<T, F>> for MatchedRoute<T
     }
 }
 
-impl<T: Clone, M, F, P> svc::Param<RouteLabels> for MatchedRoute<T, M, F, P> {
-    fn param(&self) -> RouteLabels {
-        RouteLabels(
-            self.params.parent_ref.clone(),
-            self.params.route_ref.clone(),
-        )
-    }
-}
```

Comment:
:point_up: here's the core bit driving the deferral of route label extraction, this is no longer some we can know in advance (_because there's a hostname for HTTP metrics_). instead, we use an `ExtractParam<P, T>` to get our labels upon being given a request to inspect.